modulus/sym/eq/derivatives.py

```diff
@@ -93,20 +97,46 @@ def dict_output(
     ) -> Dict[str, torch.Tensor]:
         return {diff(var_name, name): output_tensors[i] for i, name in enumerate(sizes)}
 
+    # GradScaler and DerivScalers are shared python objects which could/should not
+    # be torchscripted.
+    @torch.jit.ignore
+    def _scale(self, var: torch.Tensor, var_name: str) -> torch.Tensor:
+        return self.deriv_scalers[var_name].scale(var)
+
+    @torch.jit.ignore
+    def _unscale(self, grad: List[torch.Tensor], var_name: str) -> List[torch.Tensor]:
+        return self.deriv_scalers[var_name].unscale_deriv(grad)
+
     def forward(self, input_var: Dict[str, torch.Tensor]) -> Dict[str, torch.Tensor]:
         output_var = {}
         for var_name, grad_sizes in self.gradient_dict.items():
             var = input_var[var_name]
             grad_var = self.prepare_input(input_var, grad_sizes.keys())
-            grad = gradient(var, grad_var)
+
+            if self.scaler_enabled:
+                var = self._scale(var, var_name)
+
+            # Calculate gradient using Autodiff.
+            with torch.cuda.amp.autocast(enabled=False):
+                # Autograd within the autocast region is not recommended and could
+                # have unpredicted behavior. Autocast has thread-local configurations.
+                # Because Autograd runs in a different thread, so it could have
+                # different/default AMP dtype from the one we specified in the forward
+                # thread. Disabling autocast here will allow Autograd to run the same
+                # dtype as the forward pass.
```

Comment:
```suggestion
                    # Autocast needs to be disabled because using Autograd within the autocast region
                    # is not recommended and could have unpredictable behavior. Autocast relies on
                    # thread-local configurations, and since Autograd runs in a different thread, it
                    # could use a different/default AMP dtype than the one specified in the forward thread.
                    # Disabling autocast here will allow Autograd to run the same dtype as the forward pass.
```

Reply:
Done.